package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -408,8 +407,7 @@ def _get_dh_pairs(self):
 
         # If donors_sel is not provided, use topology to find d-h pairs
         if not self.donors_sel:
-
-            if len(self.u.bonds) == 0:
+            if not (hasattr(self.u, 'bonds') and len(self.u.bonds) != 0):
                 raise Exception('Cannot assign donor-hydrogen pairs via topology as no bonded information is present. '
```

Comment:
Can you change this to something that's not a bare exception? I am not 100% sure what it should be, possibly our own [`NoDataError`](https://www.mdanalysis.org/mdanalysis/documentation_pages/exceptions.html#MDAnalysis.exceptions.NoDataError) or maybe @richardjgowers @jbarnoud @zemanj have an idea what we typically use when bond information is missing?